db/migrate/20230202204937_add_index_to_posts_where_not_deleted_or_empty.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class AddIndexToPostsWhereNotDeletedOrEmpty < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def change
+    add_index(
+      :posts,
+      %i[id topic_id],
+      where: "deleted_at IS NULL AND raw <> ''",
+      name: "index_posts_on_id_topic_id_where_not_deleted_or_empty",
+      algorithm: :concurrently,
```

Comment:
Adding an index concurrently has the possibility of timing out leaving an invalid index behind even though the migration failed to run successfully. We can handle that by always dropping the index first before adding the index. 

Example: 

```
class AddImapGroupIdIndex < ActiveRecord::Migration[6.0]
  disable_ddl_transaction!

  def up
    execute <<~SQL
      DROP INDEX CONCURRENTLY IF EXISTS index_incoming_emails_on_imap_group_id
    SQL

    execute <<~SQL
      CREATE INDEX CONCURRENTLY index_incoming_emails_on_imap_group_id ON incoming_emails USING btree (imap_group_id)
    SQL
  end

  def down
    execute <<~SQL
      DROP INDEX IF EXISTS index_incoming_emails_on_imap_group_id
    SQL
  end
end
```